src/test/java/com/browserstack/utils/browserstack/BrowserStackUtil.java

```diff
@@ -0,0 +1,38 @@
+package com.browserstack.utils.browserstack;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class BrowserStackUtil {
+
+    private static final String BROWSERSTACK_USER_ENV = "BROWSERSTACK_USER";
+    private static final String BROWSERSTACK_ACCESS_KEY_ENV = "BROWSERSTACK_ACCESSKEY";
+    private static final String BROWSERSTACK_HUB_URL_FORMAT = "https://%s:%s@hub.browserstack.com/wd/hub";
```

Comment:
pass the user and key as test capabilities instead of passing them in the URL


browserstack.userFor running your Selenium and Appium tests on BrowserStack, it requires a username and an access key for authenticating the user. This capability can be used to set the username. | You can find your username and access key on the Settings page under the Automate section.Example: "browserstack.user": "<BROWSERSTACK_USERNAME>"
-- | --
browserstack.keyFor running your Selenium and Appium tests on BrowserStack, it requires a username and an access key for authenticating the user. This capability can be used to set the access key. | You can find your username and access key on the Settings page under the Automate section.Example: "browserstack.key": "<BROWSERSTACK_ACCESSKEY>"